BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -241,38 +250,52 @@ - (void)failAuthenticationWithConnection:(NSURLConnection *)connection
     if (self.view.window) {
-        
-        self.connectionError = error;
-        
-        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Unable to Log In", @"Alert view title: Title for failed SSO login due to authentication issue")
-                                                                                 message:message
-                                                                          preferredStyle:UIAlertControllerStyleAlert];
-        
-        __weak typeof(self) weakSelf = self;
-        void (^completion)(UIAlertAction *action) = ^void(UIAlertAction *action) {
-            [alertController dismissViewControllerAnimated:YES completion:nil];
-            
-            if (weakSelf.completionBlock) {
-                NSError *error = weakSelf.connectionError;
-                weakSelf.connectionError = nil;
-                weakSelf.completionBlock(weakSelf, nil, error);
-            }
-            // Clear out the authentication challenge in memory
-            weakSelf.authenticationChallenge = nil;
-        };
-        
-        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"OK", @"Label: Allow the user to accept the current condition, often used on buttons to dismiss alerts")
-                                                           style:UIAlertActionStyleDefault
-                                                         handler:completion];
-        
-        [alertController addAction:okAction];
-        [self presentViewController:alertController animated:YES completion:nil];
+        [self handleConnectionError];
+        // Nil out to avoid any view did appear extra call.
+        self.connectionError = nil;
+        self.connectionErrorMessage = nil;
+    } else {
+        // We still want to report the completion block so that the necessary cleanup is done up in the call tree.
+        if (self.completionBlock) {
+            self.completionBlock(self, nil, self.connectionError);
+        }
     }
+    
     [self.activityIndicator stopAnimating];
 }
 
+- (void)handleConnectionError
+{
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Unable to Log In", @"Alert view title: Title for failed SSO login due to authentication issue")
+                                                                             message:self.connectionErrorMessage
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+    
+    __weak typeof(self) weakSelf = self;
+    void (^completion)(UIAlertAction *action) = ^void(UIAlertAction *action) {
+        [alertController dismissViewControllerAnimated:YES completion:nil];
+        
+        if (weakSelf.completionBlock) {
```

Comment:
it would be the case if and only if (a very edge case imo) : the error comes when the view's window is nil, and then somehow the viewDidAppear method is called.